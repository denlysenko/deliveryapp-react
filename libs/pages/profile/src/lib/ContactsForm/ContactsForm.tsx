import React from 'react';

import { FormikHandlers, FormikErrors, FormikTouched } from 'formik';

import { InputText } from 'primereact/inputtext';
import { InputMask } from 'primereact/inputmask';
import { Message } from 'primereact/message';

import { ProfileFormValues } from '../Profile';
import { StyledForm } from '../StyledForm';

interface ContactsFormProps {
  handleChange: FormikHandlers['handleChange'];
  values: ProfileFormValues;
  errors: FormikErrors<ProfileFormValues>;
  touched: FormikTouched<ProfileFormValues>;
}

export const ContactsForm: React.FC<ContactsFormProps> = ({
  handleChange,
  values,
  errors,
  touched
}) => (
  <StyledForm>
    <h3>Contacts</h3>
    <div className="p-grid">
      <div className="p-col-12 row">
        <div className="input-wrapper p-float-label">
          <InputText
            id="firstName"
            data-testid="firstName"
            value={values.firstName}
            onChange={handleChange}
          />
          <label htmlFor="firstName">First Name</label>
          <i className="fa fa-user-circle-o"></i>
        </div>
      </div>
      <div className="p-col-12 row">
        <div className="input-wrapper p-float-label">
          <InputText
            id="lastName"
            data-testid="lastName"
            value={values.lastName}
            onChange={handleChange}
          />
          <label htmlFor="lastName">Last Name</label>
          <i className="fa fa-user-circle-o"></i>
        </div>
      </div>
      <div className="p-col-12 row">
        <div className="input-wrapper p-float-label">
          <InputText
            id="company"
            data-testid="company"
            value={values.company}
            onChange={handleChange}
          />
          <label htmlFor="company">Company</label>
          <i className="fa fa-building-o"></i>
        </div>
      </div>
      <div className="p-col-12 row">
        <div className="input-wrapper p-float-label">
          <InputText
            id="email"
            data-testid="email"
            className={touched.email && errors.email ? 'invalid' : ''}
            value={values.email}
            onChange={handleChange}
          />
          <label htmlFor="email">Email</label>
          <i className="fa fa-envelope-o"></i>
        </div>
        {touched.email && errors.email && (
          <Message
            id="email-error"
            severity="error"
            text={errors.email}
          ></Message>
        )}
      </div>
      <div className="p-col-12 row">
        <div className="input-wrapper p-float-label">
          <InputMask
            id="phone"
            mask="(999) 999-9999"
            name="phone"
            value={values.phone}
            onChange={handleChange}
          />
          <label htmlFor="phone">Phone</label>
          <i className="fa fa-phone"></i>
        </div>
      </div>
    </div>
  </StyledForm>
);
