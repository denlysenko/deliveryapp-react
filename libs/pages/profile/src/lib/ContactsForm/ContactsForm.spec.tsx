import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ContactsForm } from './ContactsForm';

const values = {
  firstName: '',
  lastName: '',
  company: '',
  email: 'test@test.com',
  phone: ''
};

const handleChange = jest.fn();

describe('[Profile] ContactsForm', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render contacts form', () => {
    const { baseElement } = render(
      <ContactsForm
        values={values}
        handleChange={handleChange}
        errors={{}}
        touched={{}}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
