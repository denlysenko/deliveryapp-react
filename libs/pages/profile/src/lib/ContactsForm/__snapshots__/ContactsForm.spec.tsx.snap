// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Profile] ContactsForm should render contacts form 1`] = `
<body>
  <div>
    <div
      class="sc-AykKC jlhElc"
    >
      <h3>
        Contacts
      </h3>
      <div
        class="p-grid"
      >
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              data-testid="firstName"
              id="firstName"
              value=""
            />
            <label
              for="firstName"
            >
              First Name
            </label>
            <i
              class="fa fa-user-circle-o"
            />
          </div>
        </div>
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              data-testid="lastName"
              id="lastName"
              value=""
            />
            <label
              for="lastName"
            >
              Last Name
            </label>
            <i
              class="fa fa-user-circle-o"
            />
          </div>
        </div>
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              data-testid="company"
              id="company"
              value=""
            />
            <label
              for="company"
            >
              Company
            </label>
            <i
              class="fa fa-building-o"
            />
          </div>
        </div>
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component p-filled"
              data-testid="email"
              id="email"
              value="test@test.com"
            />
            <label
              for="email"
            >
              Email
            </label>
            <i
              class="fa fa-envelope-o"
            />
          </div>
        </div>
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              id="phone"
              name="phone"
              type="text"
            />
            <label
              for="phone"
            >
              Phone
            </label>
            <i
              class="fa fa-phone"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
